upi/openstack/04_control-plane.yaml

```diff
@@ -43,38 +43,31 @@
     - os_networking_type == "Kuryr"
     - "os_cp_trunk_name|string not in control_plane_trunks.stdout"
 
-  - name: 'Get a token for creating the server group'
-    os_auth:
-    register: cloud
+  - name: 'List the Server groups'
+    command:
+      cmd: "openstack server group list -f json -c ID -c Name"
+    register: server_group_list
+
+  - name: 'Parse the Server group ID from existing'
+    set_fact:
+      server_group_id: "{{ (server_group_list.stdout | from_json | json_query(list_query) | first).ID }}"
+    vars:
+      list_query: "[?Name=='{{ os_cp_server_group_name }}']"
+    when:
+    - "os_cp_server_group_name|string in server_group_list.stdout"
 
-  # If microversion 2.15 is not available, replace with:
-  # - name: 'Create the Control Plane server group'
-  #   uri:
-  #     method: POST
-  #     headers:
-  #       X-Auth-Token: "{{ cloud.ansible_facts.auth_token }}"
-  #       X-OpenStack-Nova-API-Version: '2.64'
-  #     url: "{{ cloud.ansible_facts.service_catalog | selectattr('name', 'match', 'nova') | first | json_query('endpoints') | selectattr('interface', 'match', 'public') | first | json_query('url') }}/os-server-groups"
-  #     body_format: json
-  #     body:
-  #       server_group:
-  #         name: "{{ os_cp_server_group_name }}"
-  #         policy: "soft-anti-affinity"
-  #   register: cp_group
   - name: 'Create the Control Plane server group'
-    uri:
-      method: POST
-      headers:
-        X-Auth-Token: "{{ cloud.ansible_facts.auth_token }}"
-        X-OpenStack-Nova-API-Version: '2.15'
-      url: "{{ cloud.ansible_facts.service_catalog | selectattr('name', 'match', 'nova') | first | json_query('endpoints') | selectattr('interface', 'match', 'public') | first | json_query('url') }}/os-server-groups"
-      body_format: json
-      body:
-        server_group:
-          name: "{{ os_cp_server_group_name }}"
-          policies:
-          - soft-anti-affinity
-    register: cp_group
+    command:
+      cmd: "openstack --os-compute-api-version=2.15 server group create -f json -c id --policy=soft-anti-affinity {{ os_cp_server_group_name }}"
```

Comment:
Will this compute api version always be available regardless if it's osp 13 or 16? what happens if it's not?

Reply:
The maximum microversion supported in Queens (on which OSP13 [is based](https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/13/html/release_notes/chap-introduction)) [is 2.60](https://docs.openstack.org/nova/latest/reference/api-microversion-history.html). 2.15 has been introduced in Mitaka, we should have some margin :)

Reply:
As for the minimum (e.g. "will new OSP versions support old microversions?"), the documentation [says](https://docs.openstack.org/api-guide/compute/microversions.html) that they still target 2.1 as the minimum microversion. No deprecation notice in sight.

Reply:
Great!